hibernate-core/src/main/java/org/hibernate/dialect/FirebirdDialect.java

```diff
@@ -316,29 +295,6 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 		}
 
 		if ( getVersion() >= 400 ) {
-			functionRegistry.namedDescriptorBuilder( "hex_encode" )
-					.setExactArgumentCount( 1 )
-					.setInvariantType( StandardBasicTypes.STRING );
-			functionRegistry.namedDescriptorBuilder( "hex_decode" )
-					.setExactArgumentCount( 1 )
-					.setInvariantType( StandardBasicTypes.BINARY );
-			functionRegistry.namedDescriptorBuilder( "base64_encode" )
-					.setExactArgumentCount( 1 )
-					.setInvariantType( StandardBasicTypes.STRING );
-			functionRegistry.namedDescriptorBuilder( "base64_decode" )
-					.setExactArgumentCount( 1 )
-					.setInvariantType( StandardBasicTypes.BINARY );
```

Comment:
I think these conversion functions are ok to have. I guess @gavinking only meant that the internal functions `rdb$..` are an overkill

Reply:
I am against including them here unless someone is prepared to go and do the work of including them for *all* the `Dialect`s.

I'm sick of half-assed inconsistencies with this stuff.

Reply:
I don't feel strong about this, but the issue with not registering these functions is that they can't be used in the SELECT clause through the JPA `function` wrapper because the expression type is unknown. So to use this, a user would have to use a cast e.g. `cast(function('base64_encode', alias.myBlobField) as String(99999))` vs `base64_encode(alias.myBlobField)`. I just wanted to mention it because this is why I think that supporting these functions is fine. Anyway, I agree though, that it is probably a good idea to either go through all Dialects, or not include it at all.

Reply:
Right, the requirement is clear: if someone needs this stuff, they can go through and add it to all the supported `Dialects. Otherwise it's clear they don't really need it _that much_.

Reply:
Ok, lets leave them out for now